### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by huseyinbiyik**
You can find the features added for this Pull Request below. 🧩

- Remove the current_user method in ApplicationController, devise will provide us with one.
-  Install and set up devise
- Can register a new user.
- User logs in with a combination of email and password.
- Hashed passwords should be stored in the database.
- Ask for confirmation of email.
- Can reset password.
- Modify the views in devise and styling.

Coded by @huseyinbiyik  @Epaltechs 
Thank you for your review in advance! 🤓

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,7 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable, :recoverable, :rememberable, :validatable
```
Comment (Munsa1) [inline app/models/user.rb]: - [ ] Kindly add `confirmable`, as it is a requirement that the the user email must be confirmable when a user is signing up.

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @huseyinbiyik @Epaltechs, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (huseyinbiyik) [reply app/models/user.rb]: Thank you for the review and feedback @Munsa1. I fixed it now.👍

**Review by prolajumokeoni — APPROVED** [review]
Hi @huseyinbiyik, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀 
- Good design 💯 
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by huseyinbiyik** [comment]
Thank you for the review. I will merge it now 🎉

**Pull request merged**